### ISSUE TITLE
New Observation Search

### ISSUE THREAD

**Issue by kueda** [inaturalist/inaturalist#744]
Implement redesigned observation search: https://github.com/inaturalist/design/issues/24#issuecomment-148292427

- [x] prototype basic interactions using angular
  - [x] filters
    - [x] get stats to close when you hit the "Update Search" button
    - [x] add "Reset Filters" button
  - [x] stats
  - [x] view control: should toggle btwn map / grid / list, but stay in the same place for each
  - [x] map view
  - [x] grid view: https://github.com/inaturalist/design/issues/34#issuecomment-152967556
  - [x] list view: https://github.com/inaturalist/design/issues/32
- [x] enumerate any API changes required
- [x] complete pending work
  - [x] #749 new api params
  - [x] ~~#630 global firsts [can't actually remember why I added this...]~~

For production implementation, I think it would be important to keep a few things in mind:

* separate parts of the angular app into separate files: usually this means separate files for individual controllers, services, etc
* if possible, try to wrap as much UI code into directives
* testing? I skipped this in my prototyping work, and I'm not sure what the best way to structure the filesystem would be. Seems like it might be useful to test some of the filter-to-model translation, at least

**Comment by tnywlls** [comment]
Probably not in this exact bit of code, but for other searches such as searching for observations not in a project, parameters such as "not_in_project=" are not repeated when one refines the search (eg enters a taxon and searches again).  I think I've noticed this behaviour in other special-case searches (maybe fields) but can't remember them off hand.

**Comment by kueda** [comment]
@tnywlls, please stay on-topic when commenting on github issues. If you are experiencing a bug, please look for an existing issue that describes the same bug, and if you don't find one, please make a new issue. This issue is for implementing a totally new observation search UI, not for bugs in the existing search.

**Comment by kueda** [comment]
Some functional todos:

- [ ] changes to dateType should not trigger query unless actual date values have been chosen (... right?)
- [ ] clear / reset button doesn't work
- [ ] i18n strategy, maybe https://angular-translate.github.io/, maybe relying on `{{ I18n.t(key) }}`
- [ ] iNaturalistAPI/observations needs to accept a `locale` param and choose taxon names accordingly as the Rails app does, or it needs to return the `position` attribute of `TaxonName` and/or `PlaceTaxonName` associates so that the inatTaxon directive can show the right common names.

**Comment by joellebel** [comment]
everything in https://github.com/inaturalist/design/issues/61

**Comment by pleary** [comment]
I played with getting the observation snippets to show on the map on mouseover. I have an example running on the test servers. I made it so the mouseover infowindow is distinct from the one you might open by clicking on the map. That way if you have an infowindow open, and happen to mouse over the obs list then the one you opened won't go away. I was able to make it so the map doesn't move to where the window is. That means if you mouse over an obs not recorded in the current viewport, you just won't see anything. Thoughts?
